CMakeLists.txt

```diff
@@ -1199,13 +1199,17 @@ zephyr_compile_options(
   $<$<COMPILE_LANGUAGE:C>:-std=${CSTD}>
 )
 
+if(CONFIG_ARC)
 configure_file(
      $ENV{ZEPHYR_BASE}/include/arch/arc/v2/app_data_alignment.ld
      ${PROJECT_BINARY_DIR}/include/generated/app_data_alignment.ld)
+endif()
 
+if(CONFIG_ARM)
```

Comment:
app_smem_ld is also used by ARC

Reply:
then I guess `include/arch/arm/cortex_m/scripts/app_smem.ld` should be placed in some common place.